### ISSUE TITLE
`CovariantEquals` doesn't detect default `boolean equals(NotObject other)` method from an interface

### ISSUE THREAD

**Issue by PawelLipski** [checkstyle/checkstyle#12298]
```java
interface IFoo {
  default boolean equals(IFoo other) { ... }
}

class Foo extends IFoo
```

Now `Foo` has the same problem as described in https://checkstyle.sourceforge.io/config_coding.html#CovariantEquals, but Checkstyle doesn't cover it (in accordance with its own docs: `Checks that classes and records which define a covariant equals()...`).
